### PR TITLE
Avoid duplicates in path config

### PR DESCRIPTION
**Pull request by vimalk78**
Avoid duplicates in path config
 - removed duplicate in path config
 - added testcase

**Which issue(s) this PR fixes**: 
Fixes # None

**What this PR does / why we need it**: Avoid duplicates in path config

**Docs Changes**: Needed

**Release Note**:

### CODE REVIEW

**Comment by vimalk78** [comment]
how to fix the CI errors

File: lib/fluent/plugin/in_tail.rb [hunk 1]
```diff
@@ -136,7 +136,7 @@ def configure(conf)
         raise Fluent::ConfigError, "#{rc} are reserved words: #{@path_delimiter}"
       end
 
-      @paths = @path.split(@path_delimiter).map(&:strip)
+      @paths = @path.split(@path_delimiter).map(&:strip).sort().uniq()
```
Comment (ganmacs) [inline lib/fluent/plugin/in_tail.rb:139]: sort is unnecessary
```suggestion
      @paths = @path.split(@path_delimiter).map(&:strip).uniq
```

**Comment by vimalk78** [comment]
> > how to fix the CI errors
> 
> CI errors seem to have nothing with this PR. so just ignore it.
> 
> 
> `@paths` is expanded with `*` or time. So I believe that `uniq` is executed at [here](https://github.com/fluent/fluentd/blob/eaf68654d4d80071a4a149daab5579f094f8ce37/lib/fluent/plugin/in_tail.rb#L308)

The `uniq` is executed on the `@exclude_path`

**Comment by ganmacs** [comment]
Ah, you are right. then `path.uniq - excluded`
https://github.com/fluent/fluentd/blob/eaf68654d4d80071a4a149daab5579f094f8ce37/lib/fluent/plugin/in_tail.rb#L299

**Comment by vimalk78** [comment]
perhaps testcase should assert multiple TailWatchers are not created for same path

**Comment by vimalk78** [comment]
@ganmacs please check the changes.

```
  def create_driver(conf = SINGLE_LINE_CONFIG, use_common_conf = true)
    config = use_common_conf ? COMMON_CONFIG + conf : conf
    Fluent::Test::Driver::Input.new(Fluent::Plugin::TailInput).configure(config)
  end
```
while running the `test_in_tail.rb` , i found that input to `create_driver` already has `path` expanded.
If `path => /var/log/*.log`, the method will receive expanded value of path `/var/log/something/somethod.log`

So i could not fail the `expand_path` test i added. Still i hope the changes are useful.

File: lib/fluent/plugin/in_tail.rb [hunk 2]
```diff
@@ -136,7 +136,7 @@ def configure(conf)
         raise Fluent::ConfigError, "#{rc} are reserved words: #{@path_delimiter}"
       end
 
-      @paths = @path.split(@path_delimiter).map(&:strip)
+      @paths = @path.split(@path_delimiter).map(&:strip).uniq()
```
Comment (ganmacs) [inline lib/fluent/plugin/in_tail.rb:139]: unnecessary `()`
```suggestion
      @paths = @path.split(@path_delimiter).map(&:strip).uniq
```

Comment (vimalk78) [reply lib/fluent/plugin/in_tail.rb]: updated

**Comment by vimalk78** [comment]
how to fix the CI failure? need this to pass for PR merge?

**Comment by ganmacs** [comment]
No, This PR has nothing to do with the failure. Just ignore it.

**Comment by vimalk78** [comment]
Anything else i need to do get it merged?

**Comment by repeatedly** [comment]
Looks good.

**Comment by ganmacs** [comment]
Thank you 👍

**Pull request merged**